flujo/steps/cache_step.py

```
@@ -122,12 +122,38 @@ def _serialize_for_cache_key(
-                    if hasattr(obj, "__annotations__"):
-                        field_names = list(obj.__annotations__.keys())
-                    elif hasattr(obj, "__fields__"):
-                        field_names = list(obj.__fields__.keys())
-                    return {k: f"<{obj.__class__.__name__} circular>" for k in field_names}
+                    field_names: list[str] = []
+                    try:
+                        model_fields = getattr(obj.__class__, "model_fields", None)
+                        if isinstance(model_fields, dict) and model_fields:
+                            field_names = [str(k) for k in model_fields.keys()]
+                    except Exception:
+                        pass
```

Ruff (0.14.8)

`try`-`except`-`pass` detected, consider logging the exception

(S110)